### DISCUSSION TITLE
Key derivation failed (5.9.9)

### DISCUSSION THREAD

**Discussion by oleksii-iermolenko** [strongswan/strongswan#1490] [General]
Hey! 👋 
I've tried the latest version (5.9.9)

I got this error while trying to connect:
**KDF_PRF with PRF_HMAC_SHA2_256 not supported**

I've not disabled any plugins in ./configure, only --disable-kernel-netlink 

As far as I understand hmac and sha2 plugins are enabled by default, what can be causing the problem?

Thanks, Alex

**Comment by tobiasbrunner** [comment]
You need the _kdf_ plugin, which is built by default, but maybe you use a runtime config that does not load that plugin. So make sure to check strongswan.conf and the files it includes (see [here](https://docs.strongswan.org/docs/5.9/plugins/pluginLoad.html#_runtime_plugin_configuration) for details).

**Reply by oleksii-iermolenko** [reply]
Seems like the problem was in my Android.mk file
Fixed it, thanks! 🙏
